rust-symcrypt/src/cipher/mod.rs

```diff
@@ -0,0 +1,148 @@
+//! Block Cipher functions related to creating expanded keys. For further information please see symcrypt.h for more info
+//!
+//! This module provides a way to create and use Block Ciphers. Currently only AES is supported.
+//! Once an ExpandedKey is created you are able to call Encrypt or Decrypt functions depending on your desired mode
+//!
+//! # Examples
+//!
+//! ## AES Expanded Key Creation
+//! ```rust
+//!
+//! use symcrypt::cipher::AesExpandedKey;
+//! use hex::*;
+//!
+//! let key = hex::decode("5d98398b5e3b98d87e07ecf1332df4ac").unwrap();
+//! let aes_key = AesExpandedKey::new(&key).unwrap();
+//!
+//! let aes_key_clone = aes_key.clone();
+//! ```
+//!
+use crate::errors::SymCryptError;
+use crate::symcrypt_init;
+use symcrypt_sys;
+
+use std::marker::PhantomPinned;
+use std::pin::Pin;
+use std::sync::Arc;
+
+// export ciphers
+pub mod cbc;
+
+/// 16
+pub const AES_BLOCK_SIZE: u32 = symcrypt_sys::SYMCRYPT_AES_BLOCK_SIZE;
+
+/// `BlockCipherType` is an enum that enumerates all possible block ciphers that are supported.
+/// Currently the only supported type is `AesBlock`.
+pub enum BlockCipherType {
+    AesBlock,
```

Comment:
>AesBlock [](http://example.com/codeflow?start=4&length=8)

I would just call this `Aes`. AES is always a block cipher (although there are some block cipher modes that allow you to use them as stream ciphers).

Reply:
Im going to leave this one for now, changing it would be a breaking change since it would change the pub enum type which is used by GCM. WIll lump this in with the reformatting of AES-GCM